client/send-text.cc

```diff
@@ -175,6 +185,24 @@ int main(int argc, char *argv[]) {
             }
         } while (run_forever && !interrupt_received);
     }
+    else if ( rot_vertical == true) { 
+    // rotation for vertical banner text
+        if (scroll_delay_ms > 0) { 
+           do {
+              display.Fill(bg);
+              for (int s = 0; s < total_len + height && !interrupt_received; ++s) {
```

Comment:
Here, you are using total_len which is the width of the text in x-coordinates as it has been calculated above with DrawText() in the dry-run. But this doesn't really make sense here, as you are interested in the _height_.

So you need to move the dry-run into the respective if/range.